extensions/ql-vscode/src/query-history/store/variant-analysis-data-item.ts

```diff
@@ -0,0 +1,85 @@
+import { QueryLanguage } from "../../common/query-language";
+import { QueryStatus } from "../../query-status";
+import {
+  VariantAnalysisFailureReason,
+  VariantAnalysisRepoStatus,
+  VariantAnalysisStatus,
+} from "../../variant-analysis/shared/variant-analysis";
+
+// Data Model for Variant Analysis Query History Items
+// All data points are modelled, except enums.
```

Comment:
Hm, I couldn't imagine a case where we would need the content of one of the enums be restricted.

Reply:
I just tried to create a new data-enum for a different PR and found out that TS doesn't accept enum types (that are identical but with different names) as assignable to each other.

Reply:
> I couldn't imagine a case where we would need the content of one of the enums be restricted.

My reasoning here is that if we change the set of values for an enum then we might not be able to deserialize data. As a hypothetical example using real types, imagine if we deleted the `Canceled` value from the `VariantAnalysisStatus` enum. We wouldn't be able to correctly or safely read any serialized data that contained this enum value.

> I just tried to create a new data-enum for a different PR and found out that TS doesn't accept enum types (that are identical but with different names) as assignable to each other.

Hmm, you're right. That's annoying.

However, I think this is fine and I might see what's going on here. This is just a temporary workaround until we write the full data/domain mappers for the variant analysis history item.

We can fix this when we create the proper mappers, because then we can have two identical enum types and then the mappers can convert from one to the other, like we do in [`processApiRepoStatus`](https://github.com/github/vscode-codeql/blob/9015a57732d332e59ccdf961fdaa380c1348b931/extensions/ql-vscode/src/variant-analysis/variant-analysis-processor.ts#L191). Not modelling the enum types is a compromise while we don't have a proper mapper and are just relying on the types being assignable to each other.

Is that understanding correct?

Reply:
Ah, yes that makes sense!